cpp/lib/ResultSetArrow.cpp

```diff
@@ -0,0 +1,750 @@
+                    arrow::time32(arrow::TimeUnit::MILLI),
+                    arrow::default_memory_pool());
+                builder->Resize(m_totalRowCount);
+                return builder;
+            }
+            else if (scale <= 6)
+            {
+                arrow::Time64Builder * builder = new arrow::Time64Builder(
+                    arrow::time64(arrow::TimeUnit::MICRO),
+                    arrow::default_memory_pool());
+                builder->Resize(m_totalRowCount);
+                return builder;
+            }
+            else
+            {
+                arrow::Time64Builder * builder = new arrow::Time64Builder(
+                    arrow::time64(arrow::TimeUnit::NANO),
+                    arrow::default_memory_pool());
+                builder->Resize(m_totalRowCount);
+                return builder;
+            }
+            break;
+        }
+        case SF_DB_TYPE_TIMESTAMP_LTZ:
+        case SF_DB_TYPE_TIMESTAMP_NTZ:
+        case SF_DB_TYPE_TIMESTAMP_TZ:
+        {
+            if (scale == 0)
+            {
+                arrow::TimestampBuilder * builder = new arrow::TimestampBuilder(
+                    arrow::timestamp(arrow::TimeUnit::SECOND),
+                    arrow::default_memory_pool());
+                builder->Resize(m_totalRowCount);
+                return builder;
+            }
+            else if (scale <= 3)
+            {
+                arrow::TimestampBuilder * builder = new arrow::TimestampBuilder(
+                    arrow::timestamp(arrow::TimeUnit::MILLI),
+                    arrow::default_memory_pool());
+                builder->Resize(m_totalRowCount);
+                return builder;
+            }
+            else if (scale <= 6)
+            {
+                arrow::TimestampBuilder * builder = new arrow::TimestampBuilder(
+                    arrow::timestamp(arrow::TimeUnit::MICRO),
+                    arrow::default_memory_pool());
+                builder->Resize(m_totalRowCount);
+                return builder;
+            }
+            else
+            {
+                arrow::TimestampBuilder * builder = new arrow::TimestampBuilder(
+                    arrow::timestamp(arrow::TimeUnit::NANO),
+                    arrow::default_memory_pool());
+                builder->Resize(m_totalRowCount);
+                return builder;
+            }
+            break; 
+        }
+        case SF_DB_TYPE_TEXT:
+        case SF_DB_TYPE_ANY:
+        case SF_DB_TYPE_ARRAY:
+        case SF_DB_TYPE_OBJECT:
+        case SF_DB_TYPE_VARIANT:
+        default:
+        {
+            arrow::StringBuilder * builder =
+                new arrow::StringBuilder(arrow::default_memory_pool());
+            builder->Resize(m_totalRowCount);
+            return builder;
+        }
+    }
+}
+
+void Snowflake::Client::ResultSetArrow::appendChunkToBuilders(
+    std::vector<arrow::ArrayBuilder *> builders,
+    cJSON * chunk)
```

Comment:
Do we need builders? The arrow data that we receive from the Server already has the data built.

Reply:
Sorry for the late comment.
Please resolve if this has been sufficiently addressed. Otherwise, let me know if there are further issues.